staging/src/k8s.io/apiserver/pkg/endpoints/request/context.go

```diff
@@ -94,3 +98,17 @@ func AuditEventFrom(ctx context.Context) *audit.Event {
 	ev, _ := ctx.Value(auditKey).(*audit.Event)
 	return ev
 }
+
+// TraceFrom returns the value of the trace key on the ctx
+func TraceFrom(ctx context.Context) (*utiltrace.Trace, bool) {
+	trace, ok := ctx.Value(traceKey).(*utiltrace.Trace)
+	return trace, ok
+}
+
+// WithTrace returns a new trace using the provided msg and fields, nested within any trace already in the
+// provided context. Also returns a context containing the new trace
+func WithTrace(ctx context.Context, msg string, fields ...utiltrace.Field) (context.Context, *utiltrace.Trace) {
+	parent, _ := TraceFrom(ctx)
```

Comment:
Don't we need to check the second return value here? Looking above it seems like it is possibly nil.

Reply:
This is deliberate. The below `parent.Nest` can be called with a `nil` receiver. I'll add a comment explaining.

Reply:
Comment added.